api/internal/git/repospec.go

```diff
@@ -113,67 +110,67 @@ const (
 // From strings like git@github.com:someOrg/someRepo.git or
 // https://github.com/someOrg/someRepo?ref=someHash, extract
 // the parts.
-func parseGitURL(n string) (
-	host string, orgRepo string, path string, gitRef string, gitSubmodules bool, gitSuff string, gitTimeout time.Duration) {
+func parseGitURL(n string) *RepoSpec {
+	repoSpec := &RepoSpec{raw: n, Dir: notCloned, Timeout: defaultTimeout, Submodules: defaultSubmodules}
```

Comment:
Is there a reason we initialize `Timeout` and `Submodules` here?

We should set `raw` and `Dir` because they remain constant. Their initialized values are the final return values. 

On the other hand, `Timeout` and `Submodules` are always initialized on line 121 below. If they aren't present in `n`, `parseQuery` returns their default values. We are doing unnecessary work by initializing them here. Would recommend removing.

Reply:
The reason I am initialising the default values is because I think its good practice to initialize with user defined default values where possible during object creation itself.. If we create any new logic between 114 and 121 in future which will return RepoSpec we can always ensure that RepoSpect has timeout set to the default values that we want (27secs) instead of the type defaults.. That was the reasoning.. please let me know what you think

Reply:
Ah, I see. That makes sense. Thanks for the explanation!